File: 2.0/csharp/texts/sendTextsPharmacyReminders.cs
using System.Collections.Generic;
using CallfireApiClient;
using CallfireApiClient.Api.CallsTexts.Model;

public class ApiClientSample
{
    public static void Main(string[] args)
    {
        var client = new CallfireClient("api_login", "api_password");
        var recipient1 = new TextRecipient
        {
            Message = "Hello ${contact_name}, you have an appointment with Dr. Andrew "
                      + "tomorrow at 3:00 PM, if you need to reschedule please call (855)-555-4477",
            PhoneNumber = "12135551123",
            // set custom recipient attributes, they are available only to a single Call/Text
            //  action, do not confuse them with contact fields which are stored with contact
            //  and are available to each Call/Text where contact is attached to
            Attributes = new Dictionary<string, string>
            {
                {"contact_name", "Alice"},
                {"external_user_id", "45450007002"}
            }
        };
        IList<Text> texts = client.TextsApi.Send(new List<TextRecipient> {recipient1});
    }
}

File: 2.0/java/texts/sendTextsPharmacyReminders.java
import com.callfire.api.client.CallfireClient;
import com.callfire.api.client.api.callstexts.model.Text;
import com.callfire.api.client.api.campaigns.model.TextRecipient;

import java.util.Arrays;
import java.util.List;

class ApiClientSample {
    public static void main(String[] args) {
        CallfireClient client = new CallfireClient("api_login", "api_password");

        TextRecipient recipient = new TextRecipient();
        recipient.setPhoneNumber("12135551123");
        recipient.setMessage("Hello ${contact_name}, you have an appointment with Dr. Andrew "
            + "tomorrow at 3:00 PM, if you need to reschedule please call (855)-555-4477");
        // Set custom attribute
        recipient.getAttributes().put("external_user_id", "45450007002");
        recipient.getAttributes().put("contact_name":"Alice");

        List<TextRecipient> recipients = Arrays.asList(recipient);
        List<Text> texts = client.textsApi().send(recipients);
    }
}
